### PR TITLE
[Feature] Collapsable list item for Feedback

### PR DESCRIPTION
**Pull request by rs-maker**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest2.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I added multiple integration tests (Jest) related to the features (with a high test coverage)
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes

### Motivation and Context
 Feedback items have long text making the feedback view crowded
 #3217 

### Description
 added angular/material and angular/cdk to be able to use https://material.angular.io/components/expansion/overview 
which i then used to create feedback-collapse component to make feedback un-/collapsable

### Steps for Testing

0. Be logged in as instructor
1. Navigate: Course Management -> Some Course -> Exercises -> If there are none generate a basic one for testing -> Edit in Editor 
2. Now you should be able to see the feedback on the top right
3. if none of the feedback messages are over 200 characters go in the test repository and change a message to be longer 
testing_feedback
4. Save and submit
5. look at feedback

### Test Coverage

TODO

### Screenshots

feedback-collapse-final

### CODE REVIEW

**Review by willergeorg — APPROVED** [review]
Tested on TS1. Worked for me, the message is collapsed properly.

**Review by LDAP — DISMISSED** [review]
LGTM. Love the new icon :)

**Review by dfuchss — APPROVED** [review]
Code looks good :)

**Comment by krusche** [comment]
I have two small change requests:
1) can we have the element to expand / collapse feedback on the left side?
2) can you add "..." when you collapse the feedback text? Then it's more obvious for users that some text is not displayed

File: package.json [hunk 1]
```diff
@@ -78,7 +79,7 @@
         "zone.js": "0.11.4"
     },
     "devDependencies": {
-        "@angular/cdk": "11.2.9",
+        "@angular/cdk": "^11.2.9",
```
Comment (krusche) [inline package.json:82]: don't use the ^ operator

File: package.json [hunk 2]
```diff
@@ -14,6 +14,7 @@
         "@angular/core": "11.2.10",
         "@angular/forms": "11.2.10",
         "@angular/localize": "11.2.10",
+        "@angular/material": "^11.2.9",
```
Comment (krusche) [inline package.json:17]: can we please avoid the additional dependency?

Comment (krusche) [reply package.json:17]: we use bootstrap and not material in angular. Having both libraries is overhead and will lead to inconsistencies

**Review by krusche — CHANGES_REQUESTED** [review]
I have two small change requests:
1) can we have the element to expand / collapse feedback on the left side?
2) can you add "..." when you collapse the feedback text? Then it's more obvious for users that some text is not displayed

also see my inline comments

**Comment by rs-maker** [comment]
> I have two small change requests:
> 
> 1. can we have the element to expand / collapse feedback on the left side?
> 2. can you add "..." when you collapse the feedback text? Then it's more obvious for users that some text is not displayed
> 
> also see my inline comments

I will get to implementing these + your inline comments right now.

**Review by NicolasUllmann — APPROVED** [review]
Tested on server 1. Everything worked for me.

**Review by sinangunduz — DISMISSED** [review]
tested on ts1, works as in description.

**Comment by rs-maker** [comment]
Unfortunately for everyone who has tested the old version Stephan had requested some changed which i have done now. This is how it looks now.
feedback-collapse-final

File: package.json (hunk 2 above)
Comment (rs-maker) [reply package.json]: I removed this.

File: package.json (hunk 1 above)
Comment (rs-maker) [reply package.json]: also removed this.

**Review by maximilianbergmann — DISMISSED** [review]
Tested it on TS3. Everything worked. Mentioned changes worked as expected.

File: package.json [hunk 3]
```diff
@@ -9,6 +9,7 @@
     ],
     "dependencies": {
         "@angular/animations": "11.2.10",
+        "@angular/cdk": "11.2.9",
```
Comment (krusche) [inline package.json:12]: Please revert this change

Comment (rs-maker) [reply package.json]: reverted. i dont know why yarn did that.

**Review by ArtemTsurikov — APPROVED** [review]
Tested on TS1, worked fine!

**Review by PatZintel — APPROVED** [review]
Tested on ts2, Changes look as described, nothing looked suspicious

**Review by KiraNickel — APPROVED** [review]
Tested on ts3. Works as described 👍🏼

**Review by krusche — APPROVED** [review]
